src/core/regimes.rkt

```diff
@@ -22,10 +22,42 @@
 ;; pidx = Point index: The index of the point to the left of which we should split.
 (struct si (cidx pidx) #:prefab)
 
+(define (pareto-regimes sorted ctx)
+  (define branch-exprs
+    (if (flag-set? 'reduce 'branch-expressions)
+        (exprs-to-branch-on sorted ctx)
+        (context-vars ctx)))
+  (define err-lsts (batch-errors (map alt-expr sorted) (*pcontext*) ctx))
+  (define init-errs (for/hash ([bexpr branch-exprs]) (values bexpr -1)))
+  (let loop ([alts sorted] [errs init-errs] [try-first (first branch-exprs)])
+    (cond
+     [(or (null? alts) (= (length alts) 1)) '()]
+     [else
+      (define recomputed-branch-exprs
+        (if (flag-set? 'reduce 'branch-expressions)
+            (exprs-to-branch-on alts ctx)
+            (context-vars ctx)))
+      
+      (define recomp-errs 
+        (for/hash ([bexpr recomputed-branch-exprs]) (values bexpr (hash-ref errs bexpr))))
+
+      ;; if after recomputing branch-expressions, the previously chosen branch-expression no longer exists, use a
+      ;; different one.
+      (define recomp-try 
+        (if (member try-first recomputed-branch-exprs) 
+            try-first 
+            (argmin (curry hash-ref recomp-errs) recomputed-branch-exprs)))
+
+      (define-values (opt next-try new-errs) 
+        (infer-splitpoints alts recomputed-branch-exprs recomp-errs recomp-try ctx))
+      (define high (si-cidx (argmax (λ (x) (si-cidx x)) (option-split-indices opt))))
+      (cons opt (loop (take alts high) new-errs next-try))])))
+
 ;; `infer-splitpoints` and `combine-alts` are split so the mainloop
 ;; can insert a timeline break between them.
 
 (define (infer-splitpoints alts branch-exprs cerrs try-first ctx)
+  (timeline-event! 'regimes)
```

Comment:
I would just put this in `pareto-regimes`. I don't see the point in having this exact phase appear over and over again.

Reply:
If herbie is called without the pareto flag on (which these days is basically never) infer-splitpoints is called directly from mainloop. This makes sure the timeline still tracks regimes in that scenario.

Reply:
Ugh, ok. I would rather the code path be simpler (everyone calls `pareto-regimes`, maybe it takes a flag for doing just one?) but whatever. It's not important for this PR.